pandas/_libs/lib.pyx

```diff
@@ -750,13 +750,16 @@ cpdef ndarray[object] ensure_string_array(
 
     if hasattr(arr, "to_numpy"):
 
-        if hasattr(arr, "dtype") and arr.dtype.kind in "mM":
+        if (
+            hasattr(arr, "dtype")
+            and arr.dtype.kind in "mM"
+            and not hasattr(arr, "_pa_array")
+        ):
             # dtype check to exclude DataFrame
             # GH#41409 TODO: not a great place for this
             out = arr.astype(str).astype(object)
             out[arr.isna()] = na_value
             return out
-        arr = arr.to_numpy(dtype=object)
```

Comment:
The reason for this removal is that below we already do `result = np.asarray(arr, dtype="object")`, which has the same result. And keeping `arr` the original EA instead of overriding it with a np.ndarray ensures other logic of `result is arr` works as expected.